content/post/review-building-static-analysis-tools.md

```diff
@@ -0,0 +1,249 @@
+
+* other researchers use similar approach i.e in early “[Learning Natural Coding Conventions](https://arxiv.org/abs/1402.4182)” paper by [Miltos Allamanis](undefined) and [https://ml4code.github.io](https://ml4code.github.io/) group
+> We demonstrate that coding conventions are important to software teams, by showing that 1) empirically, programmers enforce conventions heavily through code review feedback and corrective commits, and 2) **patches that were based on NATURALIZE suggestions have been incorporated** into 5 of the most popular open source Java projects on GitHub — of the 18 patches that we submitted, 14 were accepted
+
+For organization it has a huge disadvantage though — it’s laborious and hard to scale. If conducted without a proper care, results will not only be ignored by developers, but can also contribute to overall issue-tracker value depreciation for the project.
+
+Despite that, one can see this approach been used by companies in this field i.e [American Software Safety Reliability Company](http://www.assrc.us), Atlanta-based enterprise that seems to have deep roots in software verifications and somehow [supported by DARPA](http://www.qbitlogic.com/darpa-bigcode/), to do archive the same — test some of their products like [https://www.mycode.ai](https://www.mycode.ai/) solution, that is planned to deploy across all of the U.S. Department of Defense software development divisions, i.e on [Git, popular OSS project](https://public-inbox.org/git/CAGm8dMApDdLEzeKU-h16G0NSpnuk9LMTWA29t4MxO1qcNpUvhA@mail.gmail.com/).
+
+### 3. 2011 Code review integration
+
+Here, the intuition was that at the code review stage, developers are preparing their code to be seen, so they are more receptive to suggestions and readability and stylistic comments.
+
+This is aligned with direction that few people at [source{d}](undefined) are [exploring now](https://github.com/campoy/goodgopher/blob/d15a7eccc3bcb2484234732e538fc425d728e972/README.md).
+There is also some number of existing solutions in the same field:
+
+* [https://codeclimate.com](https://codeclimate.com/)
+* [https://www.codacy.com](https://www.codacy.com/)
+* [https://sideci.com](https://sideci.com/)
+* [https://houndci.com](https://houndci.com/)
+* [https://github.com/haya14busa/reviewdog](https://github.com/haya14busa/reviewdog)
+
+Authors were focused on just presenting the FindBugs results at the internal code review tool as “comments”, but this attempt did not take off either. Their integration included few notable additional features:
+
+* individual developer could suppress false positives and filter the comments by confidence levels
+* it was meant to show only new FindBugs warnings, but sometimes those issues were misscategorized as new
+
+Although code review still seems to be the best time for surfacing analysis results, this attempt failure was attributed to:
```

Comment:
**the** code review

Reply:
this -> that?

Reply:
I belive article can be skipped here as in `strategy for code review` or `best practices for code review`.
 
And changing `this` to `that` would potentially imply changing `here` to `there` above, so would prefer keeping as is.